packages/client/wallets/smart-wallet/src/blockchain/wallets/EVMSmartWallet.ts

```diff
@@ -41,79 +38,47 @@ export class EVMSmartWallet {
          */
         public: PublicClient;
     };
-    private readonly sendTransactionService: SendTransactionService;
 
     constructor(
-        private readonly crossmintService: CrossmintWalletService,
-        private readonly accountClient: SmartWalletClient,
-        publicClient: PublicClient<HttpTransport>,
+        client: { public: PublicClient<HttpTransport>; wallet: SmartWalletClient },
         chain: SmartWalletChain,
-        protected readonly logger = scwLogger
+        private readonly crossmintService: CrossmintWalletService,
+        private readonly sendTransactionService = new SendTransactionService(client)
     ) {
         this.chain = chain;
-        this.client = {
-            wallet: accountClient,
-            public: publicClient,
-        };
-        this.sendTransactionService = new SendTransactionService(publicClient);
+        this.client = client;
     }
 
     /**
      * The address of the smart wallet.
      */
     public get address() {
-        return this.accountClient.account.address;
+        return this.client.wallet.account.address;
     }
 
     /**
+     * Transfers tokens from the smart wallet to a specified address.
+     * @param toAddress The recipient's address.
+     * @param config The transfer configuration, including token details and amount.
      * @returns The transaction hash.
+     * @throws {SmartWalletError} If there's a chain mismatch between this wallet and the input configuration.
+     * @throws {SendTransactionError} If the transaction fails to send. Contains the error thrown by the viem client.
+     * @throws {SendTransactionExecutionRevertedError} A subclass of SendTransactionError if the transaction fails due to a contract execution error.
      */
-    public async transferToken(toAddress: string, config: TransferType): Promise<string> {
-        return this.logger.logPerformance(
-            "TRANSFER",
```

Comment:
It's redundant to log performance here, since we already do so within the underlying viem client. The time it takes to encode the data in `transferParams` is negligible.